src/main/java/org/fundacionjala/at15/katas/langtonant/diana/Ant.java

```diff
@@ -0,0 +1,139 @@
+                if (grid.getSquare(posX, posY) == 1) { // es blanco?
+                    if (isValidNextPosition(posX, posY + 1, "right")) {
+                        grid.changeSquare(posX, posY);
+                        posY++;
+                        this.direction = "right";
+                    } else
+                        finished = true;
+                } else {
+                    if (isValidNextPosition(posX, posY - 1, "left")) {
+                        grid.changeSquare(posX, posY);
+                        posY--;
+                        this.direction = "left";
+                    } else
+                        finished = true;
+                }
+                break;
+            case "right":
+                // System.out.println("entro RIGHT");
+                if (grid.getSquare(posX, posY) == 1) {
+                    if (isValidNextPosition(posX + 1, posY, "down")) {
+                        grid.changeSquare(posX, posY);
+                        posX++;
+                        direction = "down";
+                    } else
+                        finished = true;
+                } else {
+                    if (isValidNextPosition(posX - 1, posY, "up")) {
+                        grid.changeSquare(posX, posY);
+                        posX--;
+                        direction = "up";
+                    } else
+                        finished = true;
+                }
+                break;
+
+            case "left":
+                if (grid.getSquare(posX, posY) == 1) {
+                    if (isValidNextPosition(posX - 1, posY, "up")) {
+                        grid.changeSquare(posX, posY);
+                        posX--;
+                        direction = "up";
+                    } else
+                        finished = true;
+                } else {
+                    if (isValidNextPosition(posX + 1, posY, "down")) {
+                        grid.changeSquare(posX, posY);
+                        posX++;
+                        direction = "down";
+                    } else
+                        finished = true;
+                }
+                break;
+
+            case "down":
+                if (grid.getSquare(posX, posY) == 1) {
+                    if (isValidNextPosition(posX, posY - 1, "left")) {
+                        grid.changeSquare(posX, posY);
+                        posY--;
+                        direction = "left";
+                    } else
+                        finished = true;
+                } else {
+                    if (isValidNextPosition(posX, posY + 1, "right")) {
+                        grid.changeSquare(posX, posY);
+                        posY++;
+                        direction = "right";
+                    } else
+                        finished = true;
+                    break;
+
+                }
+        }
+
+    }
+
+    public boolean isValidNextPosition(int x, int y, String dir) {
```

Comment:
For verifying a valid position, could it be enough just to check that `x` and `y` are inside the grid ranges? Something like
```
0 <= x < grid.getRows()
0 <= y < grid.getColumns()
```
`dir` might not be needed

Reply:
The observation was taken into account.